ranger-plugin/admin-plugin/src/main/java/org/apache/hawq/ranger/service/HawqClient.java

```diff
@@ -0,0 +1,314 @@
+            } finally {
+                closeResultSet(resultSet);
+                closeStatement(preparedStatement);
+                closeConnection(conn);
+            }
+        }
+        return new ArrayList<>(uniqueResults);
+    }
+
+    private List<String> queryHawq(String userInput, String columnName, String query, String database) {
+        List<String> result = new ArrayList<>();
+        Connection conn = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+
+        try {
+            conn = getConnection(connectionProperties, database);
+            preparedStatement = handleWildcardPreparedStatement(userInput, query, conn);
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("<== HawqClient.queryHawq Starting query: " + query);
+            }
+
+            resultSet = preparedStatement.executeQuery();
+
+            while(resultSet.next()) {
+                result.add(resultSet.getString(columnName));
+            }
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("<== HawqClient.queryHawq Query result: " + result.toString());
+            }
+
+        } catch (SQLException e) {
+            LOG.error("<== HawqClient.queryHawq Error: Failed to get result from query: " + query + ", Error: " + e);
+        } finally {
+            closeResultSet(resultSet);
+            closeStatement(preparedStatement);
+            closeConnection(conn);
+        }
+
+        return result;
+    }
+
+    private Connection getConnection(Map<String, String> connectionProperties) throws SQLException {
+        return getConnection(connectionProperties, null);
+    }
+
+    private Connection getConnection(Map<String, String> connectionProperties, String database) throws SQLException {
+
+        String db = database != null ? database : DEFAULT_DATABASE;
+        String url = String.format("jdbc:postgresql://%s:%s/%s", connectionProperties.get("hostname"), connectionProperties.get("port"), db);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("<== HawqClient.checkConnection Connecting to: (" + url + ") with user: " + connectionProperties.get("username") );
+        }
+
+        Properties props = new Properties();
+        props.setProperty("user", connectionProperties.get("username"));
+        props.setProperty("password", connectionProperties.get("password"));
+        return DriverManager.getConnection(url, props);
+    }
+
+    private PreparedStatement handleWildcardPreparedStatement(String userInput, String query, Connection conn) throws SQLException {
+        PreparedStatement preparedStatement = conn.prepareStatement(query);
+        preparedStatement.setString(1, userInput.equals(WILDCARD) ? "%" : (userInput + "%"));
+        return preparedStatement;
+    }
+
+    private void closeResultSet(ResultSet rs) {
+        try {
+            if (rs != null) rs.close();
+        } catch (Exception e) {
+            // ignore
```

Comment:
There is nothing we can do, this is a standard practice to ignore closing errors. We can log them, but that would not be of a great value to anybody.